app/models/rhapsody.js

```diff
@@ -1,16 +1,27 @@
 'use strict';
 
 let Bookshelf = require('app/bookshelf');
+let errors = require('app/errors');
 
 let Rhapsody = Bookshelf.Model.extend({
     tableName: 'rhapsody',
     hasTimestamps: true,
 
+    save: function () {
+        return Bookshelf.Model.prototype.save.apply(this, arguments)
+            .catch(err => {
+                if (err.code === '23505') { //unique field violiation
```

Comment:
better to define that code as a constant
and use like so
e.g `if (err.code === ERR_CODE_UNIQUE_VIOLATION_POSTGRES)`